src/shutdown/umount.c

```diff
@@ -524,6 +528,58 @@ static bool nonunmountable_path(const char *path) {
                 || path_startswith(path, "/run/initramfs");
 }
 
+static void log_umount_blockers(const char *mnt) {
+        _cleanup_closedir_ DIR *dir = opendir("/proc");
+        if (!dir)
+                return (void) log_warning_errno(errno, "opendir(/proc) failed: %m");
+
+        _cleanup_free_ char *blockers = NULL;
+
+        FOREACH_DIRENT_ALL(de, dir, break) {
+                if (!IN_SET(de->d_type, DT_DIR, DT_UNKNOWN))
+                        continue;
+
+                pid_t pid;
+                if (parse_pid(de->d_name, &pid) < 0)
+                        continue;
+
+                _cleanup_closedir_ DIR *pid_dir = xopendirat(dirfd(dir), de->d_name, 0);
+                if (!pid_dir)
+                        continue;
```

Comment:
hmm, what's the point of maintaining pid_dir here? why not just do:

```c
_cleanup_free_ char *fd_path = NULL;

fd_path = path_join(de->d_name, fd);
if (!fd_path)
        return log_oom();

fd_dir = xopendirat(dirfddir), fd_path, 0);
```

That's two syscalls less!

(I man, I usually don't care about optimizations like that, but iterating through /proc/ is so slow anyway, let's not waste syscalls in inner loops)